ptf_tests/tests/l2l3_unsupport_tbl_port_proto_action.py

```diff
@@ -0,0 +1,129 @@
+        
+        test_params = test_params_get()
+        config_json = test_params['config_json']
+        self.dataplane = ptf.dataplane_instance
+        ptf.dataplane_instance = ptf.dataplane.DataPlane(config)
+
+        self.config_data = get_config_dict(config_json)
+
+        self.gnmicli_params = get_gnmi_params_simple(self.config_data)
+        self.interface_ip_list = get_interface_ipv4_dict(self.config_data)
+
+
+    def runTest(self):
+        if not test_utils.gen_dep_files_p4c_ovs_pipeline_builder(self.config_data):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to generate P4C artifacts or pb.bin")
+
+        if not gnmi_cli_set_and_verify(self.gnmicli_params):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to configure gnmi cli ports")
+
+        ip_set_ipv4(self.interface_ip_list)
+
+        port_list = self.config_data['port_list']
+        port_ids = test_utils.add_port_to_dataplane(port_list)
+
+        for port_id, ifname in config["port_map"].items():
+            device, port = port_id
+            self.dataplane.port_add(ifname, device, port)
+
+        if not ovs_p4ctl.ovs_p4ctl_set_pipe(self.config_data['switch'], self.config_data['pb_bin'], self.config_data['p4_info']):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to set pipe")
+        
+        for table in self.config_data['table']:
+            print(f"Scenario : {table['description']}")
+            print(f"Adding {table['description']} rules")
+            if table['name'] == "pipe.ipv4_host":
+                for match_action in table['match_action']:
+                    if ovs_p4ctl.ovs_p4ctl_add_entry(table['switch'],table['name'], match_action):
+                        self.result.addFailure(self, sys.exc_info())
+                        self.fail(f"Failed: invalid entry {match_action} should not be added")
+                    print (f"EXPECTED failure to add Invalid match action {match_action}")
+            if table['name'] == "pipe.ipv6_host":
+                for match_action in table['match_action']:
+                    if ovs_p4ctl.ovs_p4ctl_add_entry(table['switch'],table['name'], match_action):
+                        self.result.addFailure(self, sys.exc_info())
+                        self.fail(f"Failed: invalid entry {match_action} should not be added")       
+            elif table['name'] == "ingress.ipv4_host_dst":
+                for match_action in table['match_action']:
+                    if not ovs_p4ctl.ovs_p4ctl_add_entry(table['switch'],table['name'], match_action):
+                        self.result.addFailure(self, sys.exc_info())
+                        self.fail(f"Failed to add table entry {match_action}")
+
+        print (f"Sending Traffic")
+        # Verify support table, ports, proto, actions
+        for table in self.config_data['table'][2:3]:
```

Comment:
same as above, role of slicing is not clear. you might want to add a comment block here for better readability.